### DISCUSSION TITLE
Feature request: Compare directories

### DISCUSSION THREAD

**Discussion by mirekjel** [mikekazakov/nimble-commander#147] [Ideas]
It's shift+F2 in TC in short. It simply select files which are not the same to the other window. They might be not the same size, older, missing (deleted) and so on. It's not meant like diff, just the selection. Imagine my son deleted 2 photos in the dir with 543 files. I have a backup and want to copy those back again. With this function it's very easy, those 2 files are selected and I just copy them. Thanks a lot, please consider that.

**Comment by mikekazakov** [comment]
That doesn't sound that difficult to implement, though the definition of "same" can be rather vague.
Might be a "good first issue".

**Reply by mirekjel** [reply]
Well, what we need to compare is:
- filename list - if the filename doesn't exist in the other panel -> select
- if it exists, is it the same size ? no -> select
- if yes, is it the same modification date ? no -> select
If the file check get to this stage it means it should be "the same" according to attributes. 
Then vice versa from the other panel to the first one.

Of course we do not compare the content, which still could be different if the file is of the same size, time stamp and filename, but this is not what this function is about. I actually use it a lot and believe me it's addictive.

**Reply by mikekazakov** [reply]
Great, thanks for the clarification!
Well if anybody is interested - looking forward to pull requests.
